diff --git a/osdialog_win.c b/osdialog_win.c
index 4c78fc4..17115ff 100644
--- a/osdialog_win.c
+++ b/osdialog_win.c
@@ -390,7 +390,7 @@ static char* osdialog_file_impl(osdialog_file_action action, const char* dir, co
 		wchar_t strFile[MAX_PATH] = L"";
 		if (filename) {
 			wchar_t* filenameW = utf8_to_wchar(filename);
-			snwprintf(strFile, MAX_PATH, L"%S", filenameW);
+			wcsncpy_s(strFile, MAX_PATH, filenameW, _TRUNCATE);
 			OSDIALOG_FREE(filenameW);
 		}
 		ofn.lpstrFile = strFile;
